pgtype/json.go

```diff
@@ -159,3 +159,24 @@ func (src *JSON) Value() (driver.Value, error) {
 		return nil, errUndefined
 	}
 }
+
+func (src *JSON) MarshalJSON() ([]byte, error) {
+	switch src.Status {
+	case Present:
+		return src.Bytes, nil
+	case Null:
+		return nil, nil
+	default:
+		return nil, errUndefined
+	}
+}
+
+func (dst *JSON) UnmarshalJSON(b []byte) error {
+	if b == nil {
+		*dst = JSON{Status: Null}
+		return nil
+	}
+
+	*dst = JSON{Bytes: b, Status: Present}
+	return nil
+}
```

Comment:
I'm not sure what the semantics of this are / should be. What does it mean to have JSON inside of JSON?

Reply:
thanks for your review; the input bytes should be the json byte array, need check; I should consider that. I think the b represent the JSON's content.
I sometimes build my struct like:
```go
type Foo struct {
   Name string `json:"name"`
   Meta pgtype.JSONB `json:"meta"`
}
```
and the struct will be serialize to json to interaction with the client, so if the struct implement the json interface, I do not write the json.Marshal; I think I have no power to contribute, sorry about that